backend/src/main/java/fk/prof/backend/model/policy/ZookeeperBasedPolicyStoreAPI.java

```diff
@@ -0,0 +1,232 @@
+            policyDetails) {
+        String zNodePath = policyPath + DELIMITER + encode(processGroup.getAppId()) + DELIMITER + encode(processGroup.getCluster()) + DELIMITER + encode(processGroup.getProcName()) + DELIMITER + POLICY_NODE_PREFIX;
+        boolean acquired;
+        try {
+            acquired = setPolicySemaphore.tryAcquire(1, TimeUnit.SECONDS);
+            if (acquired) {
+                ZookeeperUtil.writeZNodeAsync(curatorClient, zNodePath, policyDetails.toByteArray(), true, CreateMode.PERSISTENT_SEQUENTIAL).setHandler(ar -> {
+                    if (ar.succeeded()) {
+                        inMemoryPolicyCache.put(processGroup.getAppId(), processGroup.getCluster(), processGroup.getProcName(), policyDetails);
+                        setPolicySemaphore.release();
+                        future.complete();
+                    } else {
+                        setPolicySemaphore.release();
+                        future.fail(ar.cause());
+                    }
+                });
+            } else {
+                future.fail(new PolicyException("Timeout while acquiring lock while setting policy for process_group=" + processGroup, true));
+            }
+        } catch (InterruptedException e) {
+            future.fail(new PolicyException("Interrupted while acquiring lock for setting policy for process_group=" + processGroup, true));
+        } catch (Exception e) {
+            future.fail(new PolicyException("Unexpected error while setting policy for process_group=" + processGroup, true));
+        }
+    }
+
+    /**
+     * In memory replica of the ZK Policy store
+     * Created by rohit.patiyal on 15/05/17.
+     */
+    class InMemoryPolicyCache {
+        private final Map<String, Map<String, Map<String, PolicyDTO.PolicyDetails>>> processGroupAsTreeToPolicyLookup = new ConcurrentHashMap<>();
+        private Logger logger = LoggerFactory.getLogger(InMemoryPolicyCache.class);
+
+        public PolicyDTO.PolicyDetails get(String appId, String clusterId, String procName) {
+            PolicyDTO.PolicyDetails policyDetails = null;
+            try {
+                policyDetails = processGroupAsTreeToPolicyLookup.get(appId).get(clusterId).get(procName);
+            } catch (Exception ex) {
+                logger.error("No policy found for ProcessGroup : " + appId + " " + clusterId + " " + procName + " in InMemoryPolicyCache");
+            }
+            return policyDetails;
+        }
+
+        public void put(String appId, String clusterId, String procName, PolicyDTO.PolicyDetails policyDetails) {
+            processGroupAsTreeToPolicyLookup.putIfAbsent(appId, new ConcurrentHashMap<>());
+            processGroupAsTreeToPolicyLookup.get(appId).putIfAbsent(clusterId, new ConcurrentHashMap<>());
+            processGroupAsTreeToPolicyLookup.get(appId).get(clusterId).put(procName, policyDetails);
+        }
+
+        Set<String> getAppId() {
```

Comment:
Will given them explicit access mods.But since it is intended to be used by its outer class only as per current class structure, will make them private.